### PR TITLE
Add suicide tests

### PR DESCRIPTION
**Pull request by Henry-Pulver**

### CODE REVIEW

File: delta_go/check_submission.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    expected_choose_move_return_type = (int, np.int64)
+    expected_pkl_output_type = Any
+    pkl_file = load_pkl(team_name)
+
+    max_time = 5
+    t1 = time.time()
+    choose_move_no_network(
+        example_state,
+    )
+    t2 = time.time()
```
Comment (Henry-Pulver) [inline delta_go/check_submission.py:27]: Isn't this file out of date?

File: delta_go/coords.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    row_from_bottom = int(gtpc[1:])
+    return go_base.BOARD_SIZE - row_from_bottom, col
+
+
+def to_gtp(coord):
+    """Converts from a Minigo coordinate to a GTP coordinate."""
+    if coord is None:
+        return "pass"
+    y, x = coord
+    return f"{_GTP_COLUMNS[x]}{go_base.BOARD_SIZE - y}"
```
Comment (Henry-Pulver) [inline delta_go/coords.py:95]: Seem to remember we don't use most of (or _any_ of) these?

Can we cut unused code if so - will confuse users unnecessarily

File: delta_go/game_mechanics.py [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+        render: bool = False,
+        game_speed_multiplier: float = 1.0,
+    ):
+
+        self.opponent_choose_move = opponent_choose_move
+        self.render = render
+        self.verbose = verbose
+        self.game_speed_multiplier = game_speed_multiplier
+
+        # Which color do we play as
```
Comment (Henry-Pulver) [inline delta_go/game_mechanics.py:95]: This comment fills me with fear

File: delta_go/liberty_tracker.py [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+            liberty_cache=new_lib_cache,
+            max_group_id=self.max_group_id,
+        )
+
+    def add_stone(self, color, c):
+        assert self.group_index[c] == MISSING_GROUP_ID
+        captured_stones = set()
+        opponent_neighboring_group_ids = set()
+        friendly_neighboring_group_ids = set()
+        empty_neighbors = set()
```
Comment (Henry-Pulver) [inline delta_go/liberty_tracker.py:80]: Types on this whole class would be 10/10. Also an explanation of what the class is in a docstring would be useful.

As far as I can remember, the `liberty_cache` property is a np array of the number of liberties each square has (including liberties of the group that square is a part of )

File: delta_go/main.py [hunk 5]
```diff
@@ -2,14 +2,28 @@
...
 TEAM_NAME = "Team Name"  # <---- Enter your team name here!
 assert TEAM_NAME != "Team Name", "Please change your TEAM_NAME!"
 
 
+class MCTS:
+    def __init__(self):
+        """You can use this as an mcts class that persists across choose_move calls."""
+        pass
+
+    def prune(self):
```
Comment (Henry-Pulver) [inline delta_go/main.py:20]: In example code, this is `prune_tree()` and I suggest we follow suit here

File: README.md [hunk 6]
```diff
@@ -24,60 +24,59 @@ The game ends when both players pass.
 
 As in the [CGOS](http://www.yss-aya.com/cgos/) rules, we will use [area scoring](<https://en.wikipedia.org/wiki/Go_(game)#Scoring_rules>). So a player's score is the number of stones they have on the board + the number of empty territories they surround at the end of the game.
 
-The score also includes a [Komi](<https://en.wikipedia.org/wiki/Go_(game)#Komi>) which is a handicap given to black as it is advantageous to lay the first stone. We will use a komi of 7.5 which is subtracted from black's score at the end of the game.
+The score also includes a [Komi](<https://en.wikipedia.org/wiki/Go_(game)#Komi>) which is a handicap given to black as it is advantageous to lay the first stone. We will use a komi of `7.5` which is subtracted from black's score at the end of the game.
 
 # Competition Rules :scroll:
 
 1. You must build an agent to play Go using either a **reinforcement learning** algorithm or a **heuristic search algorithm** (such as monte carlo tree search :deciduous_tree:) or a mix of **both**!
```
Comment (Henry-Pulver) [inline README.md:31]:
```suggestion
1. You must build an agent to play Go using either a **reinforcement learning** algorithm or a **planning algorithm** (such as monte carlo tree search :deciduous_tree:) or a mix of **both**!
```

File: README.md [hunk 7]
```diff
@@ -24,60 +24,59 @@ The game ends when both players pass.
...
 
 2. You can only write code in `main.py`
 
+   - Your choose_move function will have a limit of 1 second to run!
    - You can only store data to be used in a competition in a `.pkl` file by `save_file()`.
-   - You can pkl anything you want, even a dictionary of pytorch networks! Just make sure your choose_move can read it.
+   - You can pkl anything you want, even a dictionary of pytorch networks (or nothing)! Just make sure your choose_move can read it.
    - In the competition, your agent will call the `choose_move()` function in `main.py` to select a move (`choose_move()` may call other functions in `main.py`)
```
Comment (Henry-Pulver) [inline README.md:38]: I think we should explain that we are initialising their `MCTS` class in the competition, so they cannot add arguments to it.

File: delta_go/check_submission.py (hunk 1 above)
Comment (Henry-Pulver) [reply delta_go/check_submission.py:27]: Plus, we want to check that we can initialise their `MCTS` class here too, right? So maybe add that to `delta-utils`?

File: delta_go/render.py [hunk 8]
```diff
@@ -0,0 +1,94 @@
+from pathlib import Path
+
+import numpy as np
```
Comment (Henry-Pulver) [inline delta_go/render.py:3]: Would add a docstring explaining that this file just renders the game visually so they don't need to understand it in full

File: delta_go/liberty_tracker.py (hunk 4 above)
Comment (neuromantic99) [reply delta_go/liberty_tracker.py:87]: Have added a docstring and a ticket for the typing

**Pull request merged**